recipes-support/salt/files/0001-version.py-merge-upstream-version.py-implementation.patch

```diff
@@ -0,0 +1,1370 @@
+From e8c87af9168e6cca15b51acd47355a5a7d2f1b1a Mon Sep 17 00:00:00 2001
+From: Alex Stewart <alex.stewart@ni.com>
+Date: Mon, 22 Jan 2024 16:06:00 -0500
+Subject: [PATCH] version.py: merge upstream version.py implementation
+
+The current version.py implementation (c. salt-3000), uses git-describe
+to build a salt version string. This produces results like
+`3000.2-32-gfa9550c7f3` - which does not conform to PEP version
+standards and so causes recent versions of setuptools to throw errors
+like...
+```
+|     raise InvalidVersion(f"Invalid version: '{version}'")
+| setuptools.extern.packaging.version.InvalidVersion: Invalid version: '3000.2-32-gfa9550c7f3'
+```
+
+Backport upstream saltstack/salt:master's verison.py implementation,
+which has been refactored to generate compliant version strings.
+
+Upstream-Status: Submitted [https://github.com/ni/salt/pull/69]
```

Comment:
I assume we'd need to revert this change once https://github.com/ni/salt/pull/69 gets merged?

Reply:
It has been merged. So I'm going to drop this PR.